cpp/monitoring/gcm/exporter.cc

```diff
@@ -157,25 +171,34 @@ void GCMExporter::CreateMetrics() const {
     metric.Add("labels", labels);
     metric.Add("typeDescriptor", desc);
 
-    UrlFetcher::Request req((
-        URL(FLAGS_google_compute_monitoring_base_url + "/metricDescriptors")));
-    req.verb = UrlFetcher::Verb::POST;
-    req.headers.insert(make_pair("Content-Type", "application/json"));
-    req.headers.insert(make_pair("Authorization", "Bearer " + bearer_token_));
-    req.body = metric.ToString();
-
-    UrlFetcher::Response resp;
-    SyncTask task(executor_);
-    VLOG(1) << "Creating metric m.Name()...";
-    VLOG(2) << req.body;
-    fetcher_->Fetch(req, &resp, task.task());
-    task.Wait();
-    // TODO(alcutter): Handle this better?
-    CHECK_EQ(util::Status::OK, task.status());
-    CHECK_EQ(200, resp.status_code);
-    VLOG(1) << "Metrics Created.";
-    VLOG(2) << resp.body;
+    do {
+      UrlFetcher::Request req((URL(FLAGS_google_compute_monitoring_base_url +
+                                   "/metricDescriptors")));
+      req.verb = UrlFetcher::Verb::POST;
+      req.headers.insert(make_pair("Content-Type", "application/json"));
+      req.headers.insert(
+          make_pair("Authorization", "Bearer " + bearer_token_));
+      req.body = metric.ToString();
+
+      UrlFetcher::Response resp;
+      SyncTask task(executor_);
+      VLOG(1) << "Creating metric m.Name()...";
+      VLOG(2) << req.body;
+      fetcher_->Fetch(req, &resp, task.task());
+      task.Wait();
+      if (!task.status().ok() || resp.status_code != 200) {
+        LOG(WARNING) << "Failed to create/update metric metadata; status: "
+                     << task.status()
+                     << ", response_code: " << resp.status_code;
+        sleep(FLAGS_google_compute_monitoring_retry_delay_seconds);
```

Comment:
Same kind of story here, both for the `SyncTask`, the `sleep`, and incrementing a counter.